src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -940,193 +1030,210 @@ void func_80B161C0(EnTa* this, PlayState* play) {
+                    EnTa_SetActionFuncWithWakeUpAnimation(this, EnTa_WaitBuyMilkOrPlayCuccoGameResponse);
                     break;
                 default:
-                    func_80B14FAC(this, func_80B16364);
+                    EnTa_SetActionFuncWithWakeUpAnimation(this, EnTa_ContinueTalkInLonLonHouse);
                     break;
             }
         }
     }
-    this->unk_2E0 &= ~1;
+    this->stateFlags &= ~TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B16608(EnTa* this, PlayState* play) {
+void EnTa_IdleAfterCuccoGameFinished(EnTa* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
         switch (this->actor.textId) {
-            case 0x2085:
-                this->actionFunc = func_80B161C0;
+            case 0x2085: // "Want to try again?" (cucco game)
+                this->actionFunc = EnTa_WaitForPlayCuccoGameResponse;
                 break;
-            case 0x2086:
-                this->actionFunc = func_80B162E8;
+            case 0x2086: // "Marry malon?"
+                this->actionFunc = EnTa_WaitForMarryMalonResponse;
                 break;
-            case 0x2088:
-                this->actionFunc = func_80B1642C;
+            case 0x2088: // "Why don't you come to work here, on this ranch"
+                this->actionFunc = EnTa_TalkAfterCuccoGameWon;
                 break;
         }
+        // Disable immediate automatic talk
         this->actor.flags &= ~ACTOR_FLAG_16;
     } else {
+        // Enable immediate automatic talk
         this->actor.flags |= ACTOR_FLAG_16;
         func_8002F2CC(&this->actor, play, 1000.0f);
     }
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B166CC(EnTa* this) {
-    s16 temp_v0 = this->blinkTimer - 1;
+void EnTa_BlinkWaitUntilNext(EnTa* this) {
+    s16 blinkTimer = this->blinkTimer - 1;
 
-    if (temp_v0 != 0) {
-        this->blinkTimer = temp_v0;
+    if (blinkTimer != 0) {
+        this->blinkTimer = blinkTimer;
     } else {
-        this->unk_2B0 = func_80B16700;
+        this->blinkFunc = EnTa_BlinkAdvanceState;
     }
 }
 
-void func_80B16700(EnTa* this) {
+void EnTa_BlinkAdvanceState(EnTa* this) {
     s16 blinkTimer = this->blinkTimer - 1;
 
     if (blinkTimer != 0) {
         this->blinkTimer = blinkTimer;
     } else {
+        // Next towards closed eyes: open -> half, half -> closed
         s16 nextEyeIndex = this->eyeIndex + 1;
-        s16 blinkTimer = 3;
 
-        if (nextEyeIndex >= blinkTimer) {
-            this->eyeIndex = 0;
-            if (this->unk_2CE > 0) {
-                this->unk_2CE--;
+        // If the eyes were already closed (nextEyeIndex >= 3),
+        // set the eyes open and either set a new blink timer.
+        // If just woken up (rapidBlinks > 0), blink a few times
+        // in quick succession before starting the normal blink
+        // cycle.
+        if (nextEyeIndex >= 3) {
```

Comment:
you can add a `_max` value to the enum as the last/max value and
```suggestion
        if (nextEyeIndex >= TALON_EYE_INDEX_MAX) {
```